### ISSUE TITLE
Textový prepis stránky

### ISSUE THREAD

**Issue by kenji1322** [ceskaexpedice/kramerius-web-client#96]
Bolo by možné pri textovom prepise stránky čerpať z datastramu ALTO? Teraz je to myslím vytiahnuté z TEXT_OCR ale momentálne uvažujeme, že tohto datastreamu sa zbavíme, keďže to isté sa nachádza aj v Alte, takže preto by bolo fajn keby sa textový prepis vytvára z ALTO datastreamu.

**Comment by zabak** [comment]
Daly by se díky tomu mj. zobrazovat na konci řádku rozdělená slova jako spojená.

**Comment by honza-rychtar** [comment]
Pokud se nepletu, tak **TEXT_OCR** se používá pro indexaci fulltextu, takže se ho nedá zbavit.
Jinak až přidáme #112, tak se může textový přepis generovat jako výběr textu z celé strany.

**Comment by filak** [comment]
Nevím sice, jak jste došli k názoru, že se potřebujete zbavit TEXT_OCR streamu, ale vygenerování textu z ALTO XML něco stojí:
1. Musí se vytáhnout z Fedory
2. Musí se zparsovat na text

Dělat tohle při každé (re)indexaci, nebo při zobrazování textu je IMHO nesmysl.

**Comment by zabak** [comment]
Jo, to je pravda. Jenže v ALTO jsou rozdělená slova na konci řádku otagovaná alternativně i jako jedno slovo. Je to sice drobnost, ale potěší. Navíc pokud bychom měli ALTO bohatší (označené nadpisy apod., nebo už ve stávajícím dávali větší váhu slovům napsaným větším písmem) tak bychom potřebovali zpracovávat ALTO. Na druhou stranu je pravda, že reindexace by pak byla ještě pomalejší než teď, tj. ten proces by se musel vhodně optimalizovat. 

Ale pozor - při zobrazování textu to smysl má, protože tam se pracuje s jednou stránkou. Z ALTO se už teď tahá informace o tom, kde na obrázku hledané slovo leží. Když se kopíruje text z výřezu stránky, taky se to bere z ALTO streamu. 

Tohle je spíš důvod proč by mělo být vše místo ve Fedoře uloženo tak, aby se k tomu dalo přistupovat opravdu rychle a zároveň indexaci paralelizovat.

**Comment by honza-rychtar** [comment]
Zbavení se streamu TEXT_OCR teď není reálné. Konce řádků jste vyřešíli opravou TEXT_OCR.
Má generování textového přepisu z ALTO teď nějakou přidanou hodnotu?
Pokud bychom generování z ALTO obohatili i o použití barvy a velikosti písma, oddělění sekcí apod., tak to by bylo fajn. Otázka je jaká je vesměs kavlita ALTO a jestli by se to vůbec používalo.

**Comment by zabak** [comment]
Generování text-ocr z ALTO není teď aktuální, zde je možné zavřít, problém předefinuju a zadám do Krameria.

**Comment by zabak** [comment]
Pokud máš zvýraznit slovo v rámci stránky, tak to slovo mohlo být rozdělené a v ALTO má dvě půlky:

<String CONTENT="KNIHA" HEIGHT="38" WIDTH="188" VPOS="691" HPOS="1213" SUBS_TYPE="HypPart1" SUBS_CONTENT="KNIHAŘENÍ."/><HYP CONTENT="­"/></TextLine>
<TextLine BASELINE="792" HEIGHT="49" WIDTH="721" VPOS="745" HPOS="457"><String CONTENT="ŘENÍ." HEIGHT="49" WIDTH="175" VPOS="745" HPOS="457" SUBS_TYPE="HypPart2" SUBS_CONTENT="KNIHAŘENÍ."/>

Tj. hledat ho v rámci stránky jako SUBS_CONTENT a zvýraznit obě půlky.

**Comment by zabak** [comment]
Je důvod aby tohle bylo odloženo a nehlasovalo se o tom?

**Comment by honza-rychtar** [comment]
@zabak Asi ne.

**Comment by zabak** [comment]
Tak prosím ohodnoťte složitost, přidal jsem to do hlasovací tabulky.

**Comment by honza-rychtar** [comment]
Indexace ALTO místo TEXT_OCR se netýká klienta, takže shrnu, co nám tady z toho zbylo.

A) pro textový přepis můžeme použít ALTO místo TEXT_OCR jako to děláme při výběru textu na stránce - jen se vybere celé alto.

B) u zvýraznění nalezených slova na stránce **chceme** slova rozdělena na konci řádku spojovat, resp. chceme zvýraznit obě části slova. 
<img width="403" alt="Snímek obrazovky 2022-08-02 v 11 05 27" src="https://user-images.githubusercontent.com/1178142/182339844-e9fbc351-03c8-42a9-90be-7e53cc85d0df.png">


C) u textového přepisu **nechceme** slova rozdělena na konci řádku spojovat, protože zachováváme konce řádku, ale naopak chceme vizuálně zobrazit rozdělovník na konci řádku, který teď v případě přepisu z ALTO chybí
<img width="465" alt="Snímek obrazovky 2022-08-02 v 11 06 57" src="https://user-images.githubusercontent.com/1178142/182338303-4bb38f0f-7ab1-4af0-89fd-3ebd866b6f02.png">

Aby fungovalo B, je potřeba též spojovat slova při indexaci, aby se ta strana nejdříve vůbec našla.
